src/libs/ascent/runtimes/flow_filters/ascent_runtime_rover_filters.cpp

```diff
@@ -96,255 +94,315 @@ RoverXRay::~RoverXRay()
 void
 RoverXRay::declare_interface(Node &i)
 {
-    i["type_name"]   = "xray";
-    i["port_names"].append() = "in";
-    i["output_port"] = "false";
+  i["type_name"] = "xray";
+  i["port_names"].append() = "in";
+  i["output_port"] = "false";
 }
 
 //-----------------------------------------------------------------------------
 bool
 RoverXRay::verify_params(const conduit::Node &params,
-                                 conduit::Node &info)
+                               conduit::Node &info)
 {
-    info.reset();
-    bool res = true;
+  info.reset();
+  bool res = true;
 
-    if(! params.has_child("absorption") ||
-       ! params["absorption"].dtype().is_string() )
-    {
-        info["errors"].append() = "Missing required string parameter 'absorption'";
-        res = false;
-    }
+  if (!params.has_child("absorption") || !params["absorption"].dtype().is_string())
+  {
+    info["errors"].append() = "Missing required string parameter 'absorption'";
+    res = false;
+  }
 
-    if(! params.has_child("filename") ||
-       ! params["filename"].dtype().is_string() )
-    {
-        info["errors"].append() = "Missing required string parameter 'filename'";
-        res = false;
-    }
+  if (!params.has_child("filename") || !params["filename"].dtype().is_string())
+  {
+    info["errors"].append() = "Missing required string parameter 'filename'";
+    res = false;
+  }
 
-    if( params.has_child("emission") &&
-       ! params["emission"].dtype().is_string() )
-    {
-        info["errors"].append() = "Optional parameter 'emission' must be a string";
-        res = false;
-    }
+  if (params.has_child("emission") && !params["emission"].dtype().is_string())
+  {
+    info["errors"].append() = "Optional parameter 'emission' must be a string";
+    res = false;
+  }
 
+  bool has_width = params.has_child("width");
+  bool has_height = params.has_child("height");
 
-    if(params.has_path("image_params"))
+  if (has_width && has_height)
+  {
+    if (!params["width"].dtype().is_number())
     {
-      if( !params.has_path("image_params/log_scale") ||
-         ! params["image_params/log_scale"].dtype().is_string() )
-      {
-          info["errors"].append() = "Missing required image parameter 'log_scale' must be a string";
-          res = false;
-      }
-
-      if( !params.has_path("image_params/min_value") ||
-         ! params["image_params/min_value"].dtype().is_number() )
-      {
-          info["errors"].append() = "Missing required image parameter 'min_value' must be a number";
-          res = false;
-      }
-
-      if( !params.has_path("image_params/max_value") ||
-         ! params["image_params/max_value"].dtype().is_number() )
+      info["errors"].append() = "Optional parameter 'width' must be an integer";
```

Comment:
Above you only check is it is a number, not necessarily an integer. You can ask for integer probably with `is_integer()`.